geopandas/tests/test_geodataframe.py

```diff
@@ -44,6 +44,16 @@ def test_df_init(self):
         assert type(self.df2) is GeoDataFrame
         assert self.df2.crs == self.crs
 
+    def test_preserve_original_dataframe(self):
+        dataframe = pd.DataFrame(
+            {"latitude": [35.5, 35.2, 34.89], "longitude": [-112.02, -111.09, -115.3]}
+        )
+        GeoDataFrame(
+            dataframe,
+            geometry=[Point(x, y) for x, y in zip(dataframe.longitude, dataframe.latitude)],
+        )
+        assert_frame_equal(dataframe, dataframe)
```

Comment:
This will always be equal :).
```suggestion
        df_check = pd.DataFrame(
             {"latitude": [35.5, 35.2, 34.89], "longitude": [-112.02, -111.09, -115.3]}
         )
        assert_frame_equal(df_check, dataframe)
```

Reply:
Oh, but it will not always be equal if the original dataframe changes which was the issue right?.. without the copy happening dataframe != dataframe but we want dataframe = dataframe even after converting it to a Geodataframe

Reply:
The way you wrote it, it will always be the same as you are comparing the same variable.

`variable == variable` is always True.

Reply:
Oh yes sorry, that was really silly.. I updated it... Thanks :)